### DISCUSSION TITLE
formState.touched does not work properly [V6]

### DISCUSSION THREAD

**Discussion by ReactGirl** [react-hook-form/react-hook-form#5106] [Q&A]
**Describe the bug**
I have this little issue when even if I click inside the component, type something, clear etc, the formState.touched is never updated with right field.

**To Reproduce**
It is very simple: you just have to register the component (I'm using the remote register approach for that, which should not be an issue) and simple printing the formState.touched for that.

I also found that the dirtyFields does not work as well but first thing first.

Pseudo code:

```javascript

// Register the field remotely 
useEffect(() => {
    register({ name: "firstName" }, { required: true });
  }, [register]);

console.log('formState.isValid', formState.isValid)  // works
console.log('formState.touched', formState.touched) // always {}

// return (<div> the usual stuff </div>)

```

**Codesandbox link (Required)**

https://codesandbox.io/s/react-hook-form-validation-onblur-onchange-forked-r5nm8?file=/src/index.js

**Expected behavior**
The formidable touched property should display the field properly touched!

**Screenshots**
![image](https://user-images.githubusercontent.com/29104515/117584604-444f1e80-b0e4-11eb-96b0-30a745cd5ae7.png)

**Desktop (please complete the following information):**

- OS: Any
- Browser: Opera, Safari, Firefox you choose
- Version: Any

**Additional context**
I hope it is not an issue in the way the register works.
Following: https://react-hook-form.com/v6/api#formState

**Comment by bluebill1049** [comment]
We don't have the functionality to manually set touched with `setValue`, perhaps you want to send a feature request instead or use `useController` or `reigster` so input will get automatically updated with touched.

**Reply by ReactGirl** [reply]
Hi @bluebill1049 I understand now that we don't have this functionality.

Is there a way to tell the that the field was touched manually then?

**Reply by thanh-nguyen-95** [reply]
You can try adding an `onBlur` event listener to the `<input>` and keep track of 'touched' inputs in a seperate state. It won't affect the `formState.touched` or `formState.dirtyFields` though if that's what you want.

Just curious, why don't you use `<input ref={register} />` instead?

**Reply by ReactGirl** [reply]
@thanh-nguyen-95 Good question. It is because I'm trying to use a controlled input from an external library.

I think the best is to use the Controller component as @bluebill1049 mentioned. I will give a try and see.

**Reply by bluebill1049** [reply]
thank you @thanh-nguyen-95 you are right 👍 

@ReactGirl `useController` should do the job for your controlled input. https://react-hook-form.com/v6/api#useController
